src/utils.js

```diff
@@ -605,6 +613,141 @@ function suggestCorrectedDateFormat(inputDate) {
+ * @param {string} endTime - The end time in the format YYYY-MM-DDTHH:MM:SSZ
+ */
+function validateDateTimeRange(startTime, endTime) {
+	// Convert formatted times to Date objects for comparison
+	const start = new Date(startTime);
+	const end = new Date(endTime);
+	const now = new Date(); // Current time
+
+	// Get the current date and calculate the date 30 days ago, both in UTC
+	const today = new Date();
+	const thirtyDaysAgo = new Date(today);
+	thirtyDaysAgo.setUTCDate(today.getUTCDate() - 30);
+
+	// Validate that logs from beyond 30 days from today are not available
+	if (start < thirtyDaysAgo || end < thirtyDaysAgo) {
+		throw new Error('Cannot get logs more than 30 days old. Adjust your time range.');
+	}
+
+	// Truncate milliseconds to ensure comparison is only done up to seconds
+	start.setMilliseconds(0);
+	end.setMilliseconds(0);
+
+	// Validate startTime < endTime
+	if (start > end) {
+		throw new Error('endTime must be greater than startTime');
+	}
+
+	// Validate that endTime is not greater than the current time (now)
+	if (end > now) {
+		throw new Error('endTime cannot be in the future. Provide a valid endTime.');
+	}
+
+	if (start.getTime() === end.getTime()) {
+		throw new Error('Minimum duration should be 1 minutes. Current duration is 0 minutes.');
+	}
+
+	// Check if the duration between start and end times is greater than 30 minutes (1800 seconds)
+	const timeDifferenceInSeconds = (end.getTime() - start.getTime()) / 1000;
+
+	if (timeDifferenceInSeconds > 1800) {
+		const hours = Math.floor(timeDifferenceInSeconds / 3600); // Hours calculation
+		const minutes = Math.floor((timeDifferenceInSeconds % 3600) / 60); // Minutes calculation
+		const seconds = timeDifferenceInSeconds % 60; // Seconds calculation
+
+		throw new Error(
+			`Max duration between startTime and endTime should be 30 minutes. Current duration is ${hours} hour${
+				hours !== 1 ? 's' : ''
+			} ${minutes} minute${minutes !== 1 ? 's' : ''} and ${seconds} second${
+				seconds !== 1 ? 's' : ''
+			}.`,
+		);
+	}
+}
+
+/**
+ * Format and validate a given date string
+ * @param {string} time - The time string in the format YYYY-MM-DDTHH:MM:SSZ
+ * @returns {string|null} The formatted and validated date string or null if invalid
+ */
+function validateDateTimeFormat(time) {
+	// Regular expression to validate the input date format YYYY-MM-DDTHH:MM:SSZ
+	const dateTimeRegex = /^(?:(\d{4})-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01])T(0[0-9]|1[0-9]|2[0-3]):([0-5]\d):([0-5]\d)Z)$/;
+
+	// Convert the Date object to ISO string and remove milliseconds
+	let timeString = time.toISOString().replace(/\.\d{3}Z$/, 'Z');
+
+	// Validate the formatted time string against the regex
+	if (!dateTimeRegex.test(timeString)) {
+		const correctedDate = suggestCorrectedDateFormat(timeString);
+		if (!correctedDate) {
+			throw Error(`Found invalid date components for ${timeString}. Check and correct the date.`);
```

Comment:
```suggestion
			throw Error(`Invalid date components in ${timeString}. Confirm the date information is correct.`);
```

i would use something like this. Please repeat for similar error messages.

Reply:
sure @jhadobe, will update all the messages